src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -0,0 +1,208 @@
+	 */
+	protected static String readApartmentFromJson(Path jsonPath) throws IOException {
+		try (BufferedReader reader = Files.newBufferedReader(jsonPath, StandardCharsets.UTF_8)) {
+			String jsonLine = reader.readLine();
+			String jsonRead = jsonLine;
+
+			while ((jsonLine = reader.readLine()) != null) {
+				jsonRead = jsonRead + "\n" + jsonLine;
+			}
+
+			reader.close();
+			LOGGER.info("Apartment has been read with success");
+
+			return jsonRead;
+		}
+	}
+
+	/**
+	 * Gets the address field from an Address JSON.
+	 *
+	 * @param jsonString <i>String</i> the Address into JSON format
+	 * @return <i>String</i> the address field
+	 */
+	public static String getAddressFromJson(String jsonString) {
+		Jsonb jsonb = JsonbBuilder.create();
+
+		final LinkedHashMap<?, ?> result = jsonb.fromJson(jsonString, LinkedHashMap.class);
+		LOGGER.info("Get address");
+
+		checkArgument(!result.get("address").toString().isEmpty() && result.containsKey("address"), "There is no field adress in the JSON file.");
+		return result.get("address").toString();
+	}
+
+	/**
+	 * Converts a JSON expression to an Apartment object.
+	 *
+	 * @param jsonPath<i>Path</i> the JSON expression to convert into Apartment
+	 *                            object
+	 * @return <i>Apartment</i> the Apartment generated
+	 * @throws IOException if the file can't be red
+	 */
+	public static Apartment jsonToApartment(Path jsonPath) throws IOException {
+		String jsonString = readApartmentFromJson(jsonPath);
+		Jsonb jsonb = JsonbBuilder.create();
+		LOGGER.info("Create Json builder");
+
+		Apartment.Builder apartBuild = jsonb.fromJson(jsonString, Apartment.Builder.class);
+		LOGGER.info("Apartment created from JSON");
+
+		return apartBuild.build();
+	}
+
+	/**
+	 * Converts a JSON expression to a list of Apartments.
+	 *
+	 * @return <i>List</i> the list of Apartments created
+	 * @throws IOException if the file doesn't exists
+	 */
+	public static List<Apartment> jsonToApartments() throws IOException {
+		return jsonToApartments(startApartment());
+	}
+
+	/**
+	 * Converts a JSON expression to a list of Apartments.
+	 *
+	 * @param jsonPath <i>Path</i> the JSON expression to convert into a list of
+	 *                 Apartments
+	 * @return <i>List</i> the list of Apartments created
+	 * @throws IOException if the file doesn't exists
+	 */
+	public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
+		String jsonString = readApartmentFromJson(jsonPath);
+		List<Apartment.Builder> apartmentsBuild;
+		List<Apartment> apartments = new ArrayList<Apartment>();
+		LOGGER.info("Create ArrayList of Apartment");
+
+		Jsonb jsonb = JsonbBuilder.create();
+		LOGGER.info("Create Json builder");
+
+		apartmentsBuild = jsonb.fromJson(jsonString, new ArrayList<Apartment.Builder>() {
+			/** The Constant serialVersionUID for apartmentsBuild. */
+			private static final long serialVersionUID = 2876323727155064950L;
```

Comment:
Cette instance ne devrait jamais être sérialisée. Cf. BP Various. (Une exception à ma hantise des `@SuppressWarnings`.)

Reply:
Fix in commit 1006b10e1fa1e47cf9ae0efc68cea42b773ad859